### PR TITLE
Fixes the flightpotion check to include new races (again)

### PR DESCRIPTION
**Pull request by Sheits**
## About The Pull Request

I made a PR that addressed this issue 2 weeks ago (#59755 ) and was asked to remove this check so it could be refactored (#59756 ), I just checked now and it was closed so I'm just gonna PR the fix I did back then since I've had people reaching out to me, asking why it doesn't work.

## Changelog
:cl:
Fix: Flight potions now work on fly people, androids and skeletons
/:cl:

### CODE REVIEW

**Comment by SmArtKar** [comment]
I suggest you delete this monstrous check and just add some sort of var to species datums

**Review by Mothblocks — CHANGES_REQUESTED** [review]
make being able to use flight potion a trait (not the inverse, notably)

**Comment by Sheits** [comment]
> make being able to use flight potion a trait (not the inverse, notably)

sure thing, I might take a while since I got my hands full and I'm mostly new to coding but ill have a go lol

**Comment by tralezab** [comment]
jesus christ why are we basing gameplay balance like this based on whether the species has a sprite or not

this is mind numbingly stupid and you should have gone with my solution (though at least this lets the new wings see use)

**Comment by Mothblocks** [comment]
your solution was the correct one then it got closed

File: code/modules/mob/living/carbon/human/species_types/android.dm [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	name = "Android"
 	id = SPECIES_ANDROID
 	say_mod = "states"
-	species_traits = list(NOBLOOD)
+	species_traits = list(NOBLOOD, TRAIT_CAN_USE_FLIGHTPOTION)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/android.dm:5]: Make this an `inherent_trait` instead, and change your if condition to just `HAS_TRAIT(exposed_carbon, TRAIT_CAN_USE_FLIGHTPOTION)`

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -323,6 +323,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_BLOODSHOT_EYES "bloodshot_eyes"
 /// This mob should never close UI even if it doesn't have a client
 #define TRAIT_PRESERVE_UI_WITHOUT_CLIENT "preserve_ui_without_client"
+/// Lets the mob use flight potions
+#define TRAIT_CAN_USE_FLIGHTPOTION "can_use_flightpotion"
```
Comment (Mothblocks) [inline code/__DEFINES/traits.dm:327]: FLIGHT_POTION

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
@tralezab are u gonna reopen the pr

File: code/__DEFINES/traits.dm (hunk 2 above)
Comment (Sheits) [reply code/__DEFINES/traits.dm:327]: yes

Comment (Mothblocks) [reply code/__DEFINES/traits.dm:327]: yes not FLIGHTPOTION

File: code/modules/mob/living/carbon/human/species_types/android.dm [hunk 3]
```diff
@@ -20,6 +20,7 @@
 		TRAIT_NOHUNGER,
 		TRAIT_LIMBATTACHMENT,
 		TRAIT_NOCLONELOSS,
+		TRAIT_CAN_USE_FLIGHT_POTION
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/android.dm]: trailing comma

File: code/modules/mob/living/carbon/human/species_types/flypeople.dm [hunk 4]
```diff
@@ -2,7 +2,12 @@
 	name = "Flyperson"
 	id = SPECIES_FLY
 	say_mod = "buzzes"
-	species_traits = list(HAS_FLESH, HAS_BONE, TRAIT_ANTENNAE, TRAIT_CAN_USE_FLIGHTPOTION)
+	species_traits = list(HAS_FLESH, HAS_BONE, TRAIT_ANTENNAE)
+	inherent_traits = (
+		TRAIT_ADVANCEDTOOLUSER,
+		TRAIT_CAN_STRIP,
+		TRAIT_CAN_USE_FLIGHT_POTION
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/flypeople.dm]: trailing comma

File: code/modules/mob/living/carbon/human/species_types/flypeople.dm [hunk 5]
```diff
@@ -2,7 +2,12 @@
 	name = "Flyperson"
 	id = SPECIES_FLY
 	say_mod = "buzzes"
-	species_traits = list(HAS_FLESH, HAS_BONE, TRAIT_ANTENNAE, TRAIT_CAN_USE_FLIGHTPOTION)
+	species_traits = list(HAS_FLESH, HAS_BONE, TRAIT_ANTENNAE)
+	inherent_traits = (
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/flypeople.dm:6]:
```suggestion
	inherent_traits = list(
```

File: code/modules/mob/living/carbon/human/species_types/humans.dm [hunk 6]
```diff
@@ -2,7 +2,12 @@
 	name = "Human"
 	id = SPECIES_HUMAN
 	default_color = "FFFFFF"
-	species_traits = list(EYECOLOR,HAIR,FACEHAIR,LIPS,HAS_FLESH,HAS_BONE, TRAIT_CAN_USE_FLIGHTPOTION)
+	species_traits = list(EYECOLOR,HAIR,FACEHAIR,LIPS,HAS_FLESH,HAS_BONE)
+	inherent_traits = (
+		TRAIT_ADVANCEDTOOLUSER,
+		TRAIT_CAN_STRIP,
+		TRAIT_CAN_USE_FLIGHT_POTION
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/humans.dm]: trailing comma

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 7]
```diff
@@ -4,7 +4,12 @@
 	id = SPECIES_LIZARD
 	say_mod = "hisses"
 	default_color = "00FF00"
-	species_traits = list(MUTCOLORS, EYECOLOR, LIPS, HAS_FLESH, HAS_BONE, TRAIT_CAN_USE_FLIGHTPOTION)
+	species_traits = list(MUTCOLORS, EYECOLOR, LIPS, HAS_FLESH, HAS_BONE)
+	inherent_traits = (
+		TRAIT_ADVANCEDTOOLUSER,
+		TRAIT_CAN_STRIP,
+		TRAIT_CAN_USE_FLIGHT_POTION
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]: trailing comma

Comment (Mothblocks) [reply code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]: look you get it, do it everywhere

**Review by Mothblocks — CHANGES_REQUESTED** [review]
just for the record if arm reopens his pr that just applies this to everything im probably going to go with that, because i do think judging it based off what has a sprite is silly

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm (hunk 7 above)
Comment (Sheits) [reply code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]: yeah i think i fixed it in the "oops" commit but thanks for the other feedback :)

**Comment by Sheits** [comment]
> just for the record if arm reopens his pr that just applies this to everything im probably going to go with that, because i do think judging it based off what has a sprite is silly

yeah no problems, thanks for all the help

**Pull request merged**